metrics/web/test/common/presentation/navigation/state/navigation_notifier_test.dart

```diff
@@ -168,18 +169,149 @@ void main() {
+      ".handleLoggedOut() clears pages when a user logs out",
       () {
-        notifier.handleAuthenticationUpdates(isLoggedIn: true);
+        notifier.handleLoggedIn();
         notifier.push(DefaultRoutes.dashboard);
 
-        notifier.handleAuthenticationUpdates(isLoggedIn: false);
+        notifier.handleLoggedOut();
 
-        final pages = notifier.pages;
-        final currentPage = pages.last;
+        expect(notifier.pages, hasLength(1));
+      },
+    );
 
-        expect(pages, hasLength(1));
-        expect(currentPage.name, isLoginPageName);
+    test(
+      ".handleLoggedOut() pushes to the login page when a user logs out",
+      () {
+        notifier.handleLoggedIn();
+        notifier.push(DefaultRoutes.dashboard);
+
+        notifier.handleLoggedOut();
+
+        final currentConfiguration = notifier.currentConfiguration;
+
+        expect(currentConfiguration.name, equals(RouteName.login));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not clear pages when a user is not logged in",
+      () {
+        notifier.push(DefaultRoutes.login);
+
+        notifier.handleLoggedOut();
+
+        expect(notifier.pages, hasLength(2));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not push to the login page when a is not logged in",
+      () {
+        final initialPagesLength = notifier.pages.length;
+
+        notifier.handleLoggedOut();
+
+        final actualPagesLength = notifier.pages.length;
+
+        expect(actualPagesLength, equals(initialPagesLength));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the dashboard route if the current redirect route is null",
+      () {
+        notifier.handleLoggedIn();
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(RouteName.dashboard));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the loading route if the current redirect route is loading",
+      () {
+        notifier.handleAppInitialized(isAppInitialized: false);
+        notifier.push(DefaultRoutes.loading);
+
+        notifier.handleLoggedIn();
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(RouteName.loading));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() clears pages before redirect",
+      () {
+        notifier.handleAppInitialized(isAppInitialized: true);
+
+        notifier.handleLoggedIn();
+
+        expect(notifier.pages, hasLength(equals(1)));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the redirect route if it is not null",
+      () {
+        final configuration = DefaultRoutes.projectGroups;
+
+        notifier.handleAppInitialized(isAppInitialized: false);
+        notifier.handleInitialRoutePath(configuration);
+        notifier.handleAppInitialized(isAppInitialized: true);
```

Comment:
🔨 
```suggestion
        notifier.handleInitialRoutePath(configuration);
```
Also, would it be great to use `push` instead of `handleInitialRoutePath` to be consistent with other tests?